### ISSUE TITLE
No GCC output coloring

### ISSUE THREAD

**Issue by Centrix14** [msys2/MINGW-packages#8600]
Hello, I recently updated Msys2 and after that, the GCC output colorization stopped working in the MinGW64 environment. Besides, there are colors in other environments. Strangely, the following response comes to `gcc-v` :
```
Using built-in specs.
COLLECT_GCC=D:\msys2\mingw64\bin\g++.exe
COLLECT_LTO_WRAPPER=D:/msys2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../gcc-10.2.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,fortran,ada,objc,obj-c++,jit --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-filesystem-ts=yes --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --disable-plugin --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev9, Built by MSYS2 project' --with-bugurl=https://github.com/msys2/MINGW-packages/issues --with-gnu-as --with-gnu-ld --with-boot-ldflags='-pipe -Wl,--dynamicbase,--high-entropy-va,--nxcompat,--default-image-base-high -Wl,--disable-dynamicbase -static-libstdc++ -static-libgcc' 'LDFLAGS_FOR_TARGET=-pipe -Wl,--dynamicbase,--high-entropy-va,--nxcompat,--default-image-base-high' --enable-linker-plugin-flags='LDFLAGS=-static-libstdc++\ -static-libgcc\ -pipe\ -Wl,--dynamicbase,--high-entropy-va,--nxcompat,--default-image-base-high\ -Wl,--stack,12582912'
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.0 (Rev9, Built by MSYS2 project)
```
from which it follows that GCC, apparently, is built under MinGW32.
However, using the `-fdiagnostics-color` flags and the `GCC_COLORS` environment variable does not help.
The problem is not so serious, but unpleasant. I will be glad of any help. Thank you for earlier.

**Comment by 1480c1** [comment]
If you add `-fdiagnostics-color=always` to always force color, does it show up?

**Comment by 1480c1** [comment]
> from which it follows that GCC, apparently, is built under MinGW32.

As a note, the `-mingw32` suffix does not necessarily mean using the 32-bit compiler, but rather is because of historical reasons, the `x86_64` is more important as it notes that it's 64-bit

**Comment by Centrix14** [comment]
> If you add -fdiagnostics-color=always to always force color, does it show up?

Yes, this corrects the situation, but only partially. Before the update, the output was colored by default.
I also tried to fix the situation with the help of aliases, but, unfortunately, it did not work out: `alias gcc="gcc -fdiagnostics-color=always"` (this didn't work in make scripts).

**Comment by 1480c1** [comment]
So at least we know that the coloring code isn't broken, so it's possible just the terminal detection is broken

**Comment by Biswa96** [comment]
Can you see the colors in CMD? Run this `C:\msys64\msys2_shell.cmd -defterm -no-start -ucrt64` then try the gcc thing.

**Comment by Centrix14** [comment]
> Can you see the colors in CMD?

Yes, this is where the colors appear. In general, colorization works in all shells except MinGW64.

**Comment by lazka** [comment]
I think our patch has to be adjusted for the cygwin stdout pipe name change

**Comment by sskras** [comment]
I think it works fine:

![image](https://github.com/msys2/MINGW-packages/assets/7887758/ae66040f-5dc0-4fc6-9f6f-cd5328ddb1ef)

Comparing to the coloring in CMD (no MinTTY involved):

![image](https://github.com/msys2/MINGW-packages/assets/7887758/f7cb7bb2-5172-45ab-ba09-68e5ac170c04)

**Comment by lazka** [comment]
(It should only be a problem with Win8.1)

But it was fixed for that in #15036, so closing.

**Comment by sskras** [comment]
> note that it should only be a problem with Win8.1

Ah, thanks for the point. Do I get it right that this has to do with ConPTY?
If yes, then theoretically Win10 builds earlier than 1809 could be affected too.
